### ISSUE TITLE
Save information about structured VTK files

### ISSUE THREAD

**Issue by MuellerSeb** [nschloe/meshio#404]
Since we now got reading routines for structured VTK meshes with PR https://github.com/nschloe/meshio/pull/399, it would be nice to conserve the information about the structured mesh.

Therefor we could add the ``dataset`` dict form

https://github.com/nschloe/meshio/blob/07a88dae1a4d8037904074f6f0ac6f20a8bf5c82/meshio/vtk_io.py#L153

to the ``Mesh`` container. The dataset dict could be initialized in the [``Mesh``](https://github.com/nschloe/meshio/blob/07a88dae1a4d8037904074f6f0ac6f20a8bf5c82/meshio/mesh.py#L4) container as:

    self.dataset = {"type": "UNSTRUCTURED_GRID"}

to be consistent.

I can come up with a PR to tackle that.

**Comment by nschloe** [comment]
Perhaps we could then also _write_ structured meshes.

**Comment by MuellerSeb** [comment]
That would be nice, since the [evtk](https://github.com/paulo-herrera/PyEVTK) package got some down-sides.

**Comment by MuellerSeb** [comment]
> Perhaps we could then also _write_ structured meshes.

Question is, which other mesh-formats provide structured meshes? And: Are they compatible?

**Comment by nschloe** [comment]
Yeah, we should probably collect formats which have structured meshes and see what their representations have in common.
